### ISSUE TITLE
Widget's click handler does not provide the position of the click

### ISSUE THREAD

**Issue by gizmomogwai** [buggins/dlangui#269]

**Comment by buggins** [comment]
workaround: use widget.mouseEvent signal instead to get more information (note: click may be initiated by keyboard Enter or Space press as well)

```
widget.mouseEvent = delegate(Widget source, MouseEvent event) {
     // use event.x, event.y, event.action
     return true;
};

```

**Comment by gizmomogwai** [comment]
thank for the tip. i am doing this right now for storing the position of the last mouse event, then in the onclick listener i take this position. still, i find this a little bit complicated.

**Comment by buggins** [comment]
Why do you need separate onClick? just use mouseEvent
if (event.action == MouseAction.MouseDown && event.button == MouseButton.Left) {
    // this is click at event.x, event.y
}

**Comment by buggins** [comment]
I believe I can close this issue

**Comment by gizmomogwai** [comment]
yes please. thanks for your help. I did not know, that the click is done that easily.
how could i distinguish between single and double clicks?

**Comment by buggins** [comment]
if (event.doubleClick) {
}

// moreover, we have triple click detection
if (event.tripleClick) {
     // implemented several days ago
}

**Comment by gizmomogwai** [comment]
sure .. i saw this nice api, but if i implement the mouse up/down thing, would i not detect myself 3 single clicks, and you would then tell me there was a tripple click in the end?
